components/lib/datatable/datatable.d.ts

```diff
@@ -1269,11 +1271,11 @@ export interface DataTableProps<TValue extends DataTableValueArray> extends Omit
     /**
      * Icon to display the current sorting status.
      */
-    sortIcon?: IconType<DataTable<TValue>> | undefined;
+    sortIcon?: IconType<DataTable<TValue>, { sortOrder: Omit<SortOrder, 'null' | 'undefined'>; sorted: boolean }> | undefined;
```

Comment:
Are we sure it shouldn't be: `{ sortOrder?: SortOrder; sorted?: boolean }` as optional or have you confirmed they ALWAYS have values?

Reply:
I tried with 3-4 example docs via running server on my local machine and It never returned undefined. But, I am not sure maybe we have some edge case. If you wish, we can defined as `?` to make sure we don't break it.

Reply:
Yeah lets do that to be safe?

Reply:
Done